ycmd.el

```diff
@@ -1462,6 +1467,40 @@ Otherwise the response is probably an exception."
     (ycmd--report-status 'parsed)
     (run-hook-with-args 'ycmd-file-parse-result-hook results)))
 
+(defun ycmd-notify-buffer-visit ()
```

Comment:
These two functions are almost the same. You could make a function i.e. `ycmd--notify-server` which takes one argument for the `event_name` and wrap the function once with `BufferVisit` and once with `BufferUnload`. Also the two functions should have the prefix `ycmd--` because the should be marked as `private`.u